src/createServer.js

```diff
@@ -1,3 +1,69 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase/convertToCase');
+
+const cases = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+const createServer = () => {
+  const server = http.createServer((req, res) => {
+    res.setHeader('Content-Type', 'application/json');
+
+    const normalizedURL = new URL(req.url, `http://${req.headers.host}`);
+    const targetCase = normalizedURL.searchParams.get('toCase');
+    const originalText = normalizedURL.pathname.slice(1);
+    const possibleErrors = validate(originalText, targetCase);
+
+    if (possibleErrors.errors.length) {
+      res.statusCode = 400;
+      res.end(JSON.stringify(possibleErrors));
+
+      return;
+    }
+
+    const { originalCase, convertedText } = convertToCase(
+      originalText, targetCase,
+    );
+
+    const result = {
+      originalCase,
+      targetCase,
+      originalText,
+      convertedText,
+    };
+
+    res.end(JSON.stringify(result));
+  });
+
+  return server;
+};
+
+function validate(text, toCase) {
+  const statusText = {
+    errors: [],
+  };
+
+  if (!text) {
+    statusText.errors.push({
+      message:
+        'Text to convert is required. '
+        + 'Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+    });
```

Comment:
You can create an object with all your default error messages and use them as reference

```suggestion
      message: errorMessages.missingText
    });
```